.circleci/cimodel/caffe2_build_definitions.py

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env python3
+
+from collections import OrderedDict
+
+import cimodel.dimensions as dimensions
+import cimodel.miniutils as miniutils
+from cimodel.conf_tree import Ver
+
+
+DOCKER_IMAGE_PATH_BASE = "308535385114.dkr.ecr.us-east-1.amazonaws.com/caffe2/"
+
+DOCKER_IMAGE_VERSION = 248
+
+
+# TODO Sort the config.yml upstream so the ubuntu configs are contiguous
+CONFIG_HIERARCHY = [
+    (Ver("ubuntu", "16.04"), [
+        (Ver("cuda", "9.0"), [
+            "py2",
+            "cmake",
+        ]),
+        (Ver("cuda", "9.1"), ["py2"]),
+        (Ver("mkl"), ["py2"]),
+    ]),
+    (Ver("ubuntu", "14.04"), [
+        (Ver("gcc", "4.8"), ["py2"]),
+    ]),
+    (Ver("ubuntu", "16.04"), [
+        (Ver("gcc", "5"), ["onnx_py2"]),
+        (Ver("cuda", "8.0"), ["py2"]),
+    ]),
+    (Ver("ubuntu", "14.04"), [
+        (Ver("gcc", "4.9"), ["py2"]),
+    ]),
+    (Ver("ubuntu", "16.04"), [
+        (Ver("clang", "3.8"), ["py2"]),
+        (Ver("clang", "3.9"), ["py2"]),
+        (Ver("clang", "7"), ["py2"]),
+        (Ver("android"), ["py2"]),
+    ]),
+    (Ver("centos", "7"), [
+        (Ver("cuda", "9.0"), ["py2"]),
+    ]),
+    (Ver("macos", "10.13"), [
+        (Ver("ios"), ["py2"]),
+        (Ver("system"), ["py2"]),
```

Comment:
nit: ios and system aren't related. system qualifies where the python comes from (use the system python instead of homebrew or anaconda)